Tasks/DotNetCoreCLIV2/restorecommand.ts

```diff
@@ -17,13 +17,9 @@ export async function run(): Promise<void> {
     try {
         packagingLocation = await pkgLocationUtils.getPackagingUris(pkgLocationUtils.ProtocolType.NuGet);
     } catch (error) {
-        tl.debug('Unable to get packaging URIs, using default collection URI');
+        tl.debug('Unable to get packaging URIs');
         logError(error);
-        const collectionUrl = tl.getVariable('System.TeamFoundationCollectionUri');
-        packagingLocation = {
-            PackagingUris: [collectionUrl],
-            DefaultPackagingUri: collectionUrl
-        };
+        throw new Error(error);
```

Comment:
I'm not as much of a TypeScript/JavaScript expert as others on the team, but wrapping the exception in a new Error object seems suspicious. I'd think you would want to simply rethrow the same object in these cases:

```suggestion
        throw error;
```

Same comment in the exception-rethrowing spots too

Reply:
That makes sense, didnt think about that, @magleaso @satbai any thoughts on this?

Or we can define a string something like [this ](https://github.com/microsoft/azure-pipelines-tasks/blob/7bf2391069f629f64fc18f4258995565283ce37b/Tasks/PipAuthenticateV0/task.json#L83) in each task?

Reply:
Jonathan's comment makes sense to me. Not sure if that messages property is for localization. If not, I don't think it needs to be changed. If the message is only used in one place per task, it's just more readable to put the string in directly.

Reply:
Since we are already logging the error, I will just change it to `throw error` like Jonathan suggested

Reply:
Fixed